resources/assets/components/UserOverview/index.js

```diff
@@ -1,149 +1,85 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-import { map, keyBy } from 'lodash';
-import { RestApiClient } from '@dosomething/gateway';
+import gql from 'graphql-tag';
+import { map } from 'lodash';
+import { useQuery } from '@apollo/react-hooks';
 
 import Empty from '../Empty';
 import SignupCard from '../SignupCard';
 import MetaInformation from '../MetaInformation';
-import UserInformation from '../Users/UserInformation';
-
-class UserOverview extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.api = new RestApiClient();
-
-    this.state = {
-      loading: false,
-      signups: [],
-      campaigns: [],
-    };
-  }
-
-  componentDidMount() {
-    this.setState({
-      loading: true,
-    });
-
-    // Get user activity.
-    this.getUserActivity(this.props.user.id)
-      // Then get the campaign data tied to that activity.
-      .then(() => {
-        const ids = map(this.state.signups, 'campaign_id');
-        this.getCampaigns(ids);
-
-        this.setState({
-          loading: false,
-        });
-      });
+import UserInformation, {
+  UserInformationFragment,
+} from '../Users/UserInformation';
+
+const USER_OVERVIEW_QUERY = gql`
+  query UserOverviewQuery($id: String!) {
+    user(id: $id) {
+      ...UserInformation
+      permalink
+      source
+    }
+
+    signups(userId: $id, orderBy: "created_at,desc") {
+      id
+      quantity
+      whyParticipated
+
+      campaign {
+        id
+        internalTitle
+        startDate
+      }
+
+      posts {
+        id
+        type
+        url(w: 200, h: 200)
+      }
+    }
   }
 
-  /**
-   * Gets the user activity for the specified user and update state.
-   *
-   * @param {String} id
-   * @return {Object}
-   */
-  getUserActivity(id) {
-    const request = this.api.get('api/v3/signups', {
-      filter: {
-        northstar_id: id,
-      },
-      include: 'posts',
-      orderBy: 'created_at,desc',
-      limit: 80,
-    });
+  ${UserInformationFragment}
+`;
 
-    return request.then(result => {
-      this.setState({
-        signups: result.data,
-      });
-    });
-  }
-
-  /**
-   * Gets campaigns associated with signups.
-   *
-   * @param {Array} ids
-   * @return {Object}
-   */
-  getCampaigns(ids) {
-    this.api
-      .get('api/v3/campaigns', {
-        filter: {
-          id: ids.join(),
-        },
-        // @TODO: update this to paginate calls to smaller batches or use GraphQL
-        // to be able to request only the subset of campaign fields we actually use for the UI
-        // (to make sure performance isn't hit).
-        limit: 100,
-      })
-      .then(json =>
-        this.setState({
-          campaigns: keyBy(json.data, 'id'),
-        }),
-      );
-  }
-
-  render() {
-    const user = this.props.user;
+const UserOverview = ({ id }) => {
+  const { loading, error, data } = useQuery(USER_OVERVIEW_QUERY, {
```

Comment:
Note that this is following [the "modern" Apollo Client hooks API](https://www.apollographql.com/docs/react/data/queries/#executing-a-query), so looks a little different than usual!